src/components/Advertisment.js

```diff
@@ -109,7 +111,26 @@ const Advertisment = ({
         });
       }
     }
-  });
+  }, [
```

Comment:
I believe the canonical way to make sure an effect is only executed once is to give it an empty array of dependencies.

Reply:
Which would then make the state variable unnecessary

Reply:
I tested this, but the ads still re-initialized

Reply:
That's really weird. I'll have to do some testing tomorrow

Reply:
I don't know how you tested this, but I just used this method and in both Firefox and Chromium, the effect was run only once per page - I was able to write an article and go through a whole draft without the effect code rerunning again. Are you sure you weren't just seeing multiple ads being loaded once?

I submitted a cleaned-up version of my solution in #2135, you can try it out for yourself.